### PR TITLE
Fix minor issues

### PR DESCRIPTION
**Pull request by prateeksultania**

### CODE REVIEW

File: src/com/directi/training/codesmells/refactored/chess/ChessBoard.java [hunk 1]
```diff
@@ -125,8 +125,6 @@ public boolean isValidMove(Position from, Position to)
     // by extracting out code for knight in a separate method.
     private boolean hasNoPieceInPath(Position from, Position to)
     {
-        if (getPiece(from) instanceof Knight)
-            return hasNoPieceInPathOfKnight(from, to);
```
Comment (harshmanocha) [inline src/com/directi/training/codesmells/refactored/chess/ChessBoard.java]: We would still need to return true if it is a Knight since rest of the conditions won't apply.

File: src/com/directi/training/codesmells/refactored/chess/ChessBoard.java [hunk 2]
```diff
@@ -96,17 +98,16 @@ public boolean isEmpty(Position position)
...
     private Cell getCell(Position position)
     {
-        return _board[position.getRow()][position.getColumn()];
+        return _board[position.row][position.column];
     }
 
     //Dead-Code Code Smell fixed by removing getPlayerName and printMove methods (and also toString of Position),
     // as well as player 1 and player 2 fields.
 
+    //Fixed Duplicate Code across classes code smell by calling isEmpty method
```
Comment (prateeksultania) [inline src/com/directi/training/codesmells/refactored/chess/ChessBoard.java]: across classes sound wrong. Either we can omit it or change it to across class/ across methods of a class. Same change is required in readme.

File: src/com/directi/training/codesmells/smelly/Position.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
+package com.directi.training.codesmells.smelly;
+
+public class Position
+{
+    public final int row, column;
```
Comment (prateeksultania) [inline src/com/directi/training/codesmells/smelly/Position.java]: The general pattern followed for pojos is to have private fields and expose getters and setters if required. So we should revert the change. I had not followed this in the Pair class because I was just mimicking how the sun provided pair class was operating but yes we can make the fields private in that class also and expose getters.

File: src/com/directi/training/codesmells/refactored/Direction.java [hunk 4]
```diff
@@ -2,11 +2,21 @@
 
 public class Direction
 {
-    public final int rowOffset;
-    public final int columnOffset;
+    private int _rowOffset;
```
Comment (prateeksultania) [inline src/com/directi/training/codesmells/refactored/Direction.java]: fields can still be final

**Comment by prateeksultania** [comment]
Good to merge
